src/plugins/intel_gna/src/transformations/reorder_activation_and_pooling.cpp

```diff
@@ -40,10 +40,6 @@ ReorderActivationAndPooling::ReorderActivationAndPooling() {
         auto pool = std::dynamic_pointer_cast<ngraph::opset7::MaxPool>(pool_node);
         IE_ASSERT(pool != nullptr);
         auto kernel_shape = pool->get_kernel();
-        if (kernel_shape.size() > 1 && kernel_shape[0] > 1 && kernel_shape[1] > 1) {
-            return false;
-        }
-
```

Comment:
> Why is it being removed?

Because it breaks logic of reordering pooling and activation and leads to test failure due to wrong insertion of Identity, which is really not needed (because actual activation is exists aready). Do you know any reason why pooling and activation should not be reordered?

Reply:
> I'd like to understand why was this code here in the first place before we remove it. Do we know it?

I suppose logic was moved from legacy pass. Why it was done in legacy I don't know. But I don't see any reasons for doing it.